core/shared/src/main/scala/io/circe/Decoder.scala

```diff
@@ -112,6 +112,20 @@ trait Decoder[A] extends Serializable { self =>
       a <- this(a1)
       b <- x(a2)
     } yield (a, b)
+
+  /**
+   * Create a new decoder that performs some operation on the incoming JSON before decoding.
+   */
+  final def prepare(f: HCursor => ACursor): Decoder[A] = Decoder.instance(c => self.tryDecode(f(c)))
+
+  /**
+   * Create a new decoder that performs some operation on the result if this one succeeds.
+   *
+   * @param f a function returning either a value or an error message
+   */
+  final def emap[B](f: A => Xor[String, B]): Decoder[B] = Decoder.instance(c =>
```

Comment:
yes, you can still create a DecodingFailure without the cursor but then the cursor history would be empty